functests/performance/cpu_management.go

```diff
@@ -0,0 +1,192 @@
+		Expect(err).ToNot(HaveOccurred())
+		Expect(workerRTNodes).ToNot(BeEmpty())
+		workerRTNode = &workerRTNodes[0]
+
+		isolatedCPU = getIsolatedCpuFromMachineConfig()
+		Expect(isolatedCPU).ToNot(BeEmpty())
+		listIsolatedCPU, err = CPUListConverter(isolatedCPU)
+		Expect(err).ToNot(HaveOccurred())
+
+		reservedCPU = getReservedCpuFromKubeletConfig()
+		Expect(reservedCPU).ToNot(BeEmpty())
+		listReservedCPU, err = CPUListConverter(reservedCPU)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Describe("Verification of configuration on the worker node", func() {
+
+		It("Verify CPU reservation on the node", func() {
+			By(fmt.Sprintf("Allocatable CPU should be less then capacity by %d", len(listReservedCPU)))
+			capacityCpu, _ := workerRTNode.Status.Capacity.Cpu().AsInt64()
+			allocatableCpu, _ := workerRTNode.Status.Allocatable.Cpu().AsInt64()
+			Expect(capacityCpu - allocatableCpu).To(Equal(int64(len(listReservedCPU))))
+		})
+
+		It("Verify CPU affinity mask, CPU reservation and CPU isolation on worker node", func() {
+			By("checking isolated CPU")
+			cmd := []string{"cat", "/sys/devices/system/cpu/isolated"}
+			Expect(execCommandOnWorker(cmd, workerRTNode)).To(Equal(isolatedCPU))
+
+			By("checking reserved CPU in kubelet config file")
+			cmd = []string{"cat", "/rootfs/etc/kubernetes/kubelet.conf"}
+			Expect(execCommandOnWorker(cmd, workerRTNode)).To(ContainSubstring(fmt.Sprintf(`"reservedSystemCPUs":"%s"`, reservedCPU)))
+
+			By("checking CPU affinity mask for kernel scheduler")
+			cmd = []string{"/bin/bash", "-c", "taskset -pc $(pgrep rcu_sched)"}
+			Expect(execCommandOnWorker(cmd, workerRTNode)).To(ContainSubstring(fmt.Sprintf("current affinity list: %s", reservedCPU)))
+		})
+
+	})
+
+	Describe("Verification of cpu manager functionality", func() {
+		var testpod *corev1.Pod
+
+		AfterEach(func() {
+			err := testclient.Client.Delete(context.TODO(), testpod)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = pods.WaitForDeletion(testclient.Client, testpod, 60*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		table.DescribeTable("Verify CPU usage by stress PODs", func(guaranteed bool) {
+			listCPU := listReservedCPU
+			testpod = getStressPod(workerRTNode.Name)
+			testpod.Namespace = testutils.NamespaceTesting
+			if guaranteed {
+				listCPU = listIsolatedCPU
+				testpod.Spec.Containers[0].Resources.Limits = map[corev1.ResourceName]resource.Quantity{
+					corev1.ResourceCPU:    resource.MustParse("1"),
+					corev1.ResourceMemory: resource.MustParse("1Gi"),
+				}
+			}
+			err := testclient.Client.Create(context.TODO(), testpod)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = pods.WaitForCondition(testclient.Client, testpod, corev1.PodReady, corev1.ConditionTrue, 60*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+
+			cmd := []string{"/bin/bash", "-c", "ps -o psr $(pgrep -n stress) | tail -1"}
+			Expect(strings.Trim(execCommandOnWorker(cmd, workerRTNode), " ")).To(BeElementOf(listCPU))
```

Comment:
I can see that you are checking vs `listReservedCPU` for guaranteed and non-guaranteed pods, does it by intent?

Reply:
hm, not clear what do you mean. If it is non-guaranteed pod **listCPU := listReservedCPU**. If pod is guaranteed **listCPU = listIsolatedCPU**

Reply:
missed this variable under the `if` condition